reader_concurrency_semaphore.cc

```diff
@@ -1265,7 +1274,7 @@ bool reader_concurrency_semaphore::has_available_units(const resources& r) const
 }
 
 bool reader_concurrency_semaphore::all_need_cpu_permits_are_awaiting() const {
-    return _stats.need_cpu_permits == _stats.awaits_permits;
+    return (_stats.need_cpu_permits - _stats.awaits_permits) < _cpu_concurrency();
 }
```

Comment:
now that concurrency can be higher than 1, should we rename all_need_cpu_permits_are_awaiting() to something like room_for_new_cpu_permit()?

for my education, even permits awaiting for I/O are still accounted in need_cpu_permits, right? so we won't have an underflow here.

Reply:
Right, the name of this function doesn't look apporpriate.

Reply:
Fixed in v2.